probe-rs/src/config/registry.rs

```diff
@@ -0,0 +1,84 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+use super::target::Target;
+use super::chip::Chip;
+use super::flash_algorithm::FlashAlgorithm;
+use crate::collection::get_core;
+
+pub enum TargetSelectionError {
+    ChipNotFound,
+    VariantNotFound,
+    AlgorithmNotFound,
+    CoreNotFound,
+}
+
+pub struct Registry {
+    /// All the available chips.
+    /// <chip_name, chip>
+    chips: HashMap<String, Chip>,
+    flash_algorithms: HashMap<String, FlashAlgorithm>,
+}
+
+impl Registry {
+    pub fn load_from_dir(path: &Path) -> Registry {
```

Comment:
There's like 8 warnings in CI about `path` being unused, any idea why they repeat so often?

```suggestion
    pub fn load_from_dir(_path: &Path) -> Registry {
```

Reply:
No idea about the repeated warning. I just sometimes see repeated wanrings when running tests I think and then it is only twice.